pyingest/parsers/jats.py

```diff
@@ -142,218 +143,16 @@ def parse(self, input_data, **kwargs):
-        affils = OrderedDict()
-
-        # Author notes/note ids
-        try:
-            notes = article_meta.find('author-notes').find_all('fn')
-        except Exception as err:
-            pass
-        else:
-            for n in notes:
-                try:
-                    n.label.decompose()
-                except Exception as err:
-                    pass
-                else:
-                    try:
-                        n['id']
-                    except Exception as err:
-                        pass
-                        # print "I'm failing on author notes!",err
-                    else:
-                        key = n['id']
-                        note_text = self._detag(n, JATS_TAGSET['affiliations'])
-                        affils[key] = note_text.strip()
-
-        # Affils/affil ids
-        l_need_affils = False
-        try:
-            affil = article_meta.find('contrib-group').find_all('aff')
-            if len(affil) == 0:
-                try:
-                    affil = article_meta.find_all('aff')
-                except Exception as err:
-                    pass
-        except Exception as err:
-            pass
-        else:
-            for a in affil:
-                try:
-                    a.label.decompose()
-                except Exception as err:
-                    pass
-                try:
-                    a['id']
-                except Exception as err:
-                    l_need_affils = True
-                    pass
-                else:
-                    key = a['id']
-                    ekey = ''
-                    try:
-                        email_array = []
-                        email_a = a.find_all('ext-link')
-                        for em in email_a:
-                            if em['ext-link-type'] == 'email':
-                                address = self._detag(em, (
-                                    JATS_TAGSET['affiliations']))
-                                address_new = "<EMAIL>" + address + "</EMAIL>"
-                                ekey = em['id']
-                                if ekey is not '':
-                                    affils[ekey] = address_new
-                        while a.find('ext-link') is not None:
-                            a.find('ext-link').extract()
-                    except Exception as err:
-                        pass
-
-                    aff_text = self._detag(a, JATS_TAGSET['affiliations'])
-                    affils[key] = aff_text.strip()
-
-
-        # <contrib-group>: Author name and affil/note lists:
+        # Authors and their Affiliations:
         try:
-            authors = article_meta.find('contrib-group').find_all('contrib')
+            auth_affil = JATSContribs(soup=article_meta)
+            auth_affil.parse()
+            aa_output = auth_affil.output
         except Exception as err:
-            pass
+            print('oopsie, problem', err)
```

Comment:
Leaving it just as a comment since I don't know the status of logging/error reporting for this particular module: in general, I think it's always a good practice to print error messages to sys.stderr so that unix pipes which use stdin/stdout to communicate don't get unexpected input/output.  Using python logging would be better, with proper initializations to stderr.